### DISCUSSION TITLE
Accepting 2 connections but only connecting 1

### DISCUSSION THREAD

**Discussion by jcgomez11** [miguelgrinberg/python-socketio#912] [Q&A] [answered]
I'm deploying a server in a Raspberry Pi using eventlet and a WSGI application. When connecting a client for the first time, always 2 'accepted' messages are prompted in the terminal each with a different port number and only 1 time the program calls the connect callback for the server.  The Client is just creating 1 socket and trying to connect it.

2022-03-27 17:19:57,242 [INFO] (8007) wsgi starting up on http://0.0.0.0:1234
_2022-03-27 17:20:08,106 [DEBUG] (8007) accepted ('clientIP', 53148)_
2022-03-27 17:20:08,112 [INFO] 172.23.77.74 - - [27/Mar/2022 17:20:08] "GET /socket.io/?transport=polling&EIO=4&t=1650354994.6890473 HTTP/1.1" 200 278 0.002304
_2022-03-27 17:20:08,128 [INFO] Connected, H7lyc4i_bwxVgL-wAAAB]_
2022-03-27 17:20:08,129 [INFO] 172.23.77.74 - - [27/Mar/2022 17:20:08] "POST /socket.io/?transport=polling&EIO=4&sid=3P0tL-0LjnJV0XOaAAAA HTTP/1.1" 200 167 0.003475
_2022-03-27 17:20:08,131 [DEBUG] (8007) accepted ('clientIP', 53149)_
2022-03-27 17:20:08,133 [INFO] 172.23.77.74 - - [27/Mar/2022 17:20:08] "GET /socket.io/?transport=polling&EIO=4&sid=3P0tL-0LjnJV0XOaAAAA&t=1650354994.74309 HTTP/1.1" 200 213 0.000429 

(I changed the client IP for security reasons of the company )

There are some ghost connections happening on the server but I can't figure out why and eventually the server is crashing. This connections are coming from the client IP addres but they never show up as connected in the callback on the client. 

The setup of python and the libraries was already done when i got to troubleshoot the program and it is a little bit off, this might be one of the reasons it is behaving so extrangely.

Windows-Client
Python 3.9.4
       python-socketio 5.4.0
       python-engineio 4.2.1
       eventlet 0.32.0

Raspi-Server
Python 3.7.4
       python-socketio 5.5.2
       python-engineio 4.31
       eventlet 0.20.0

Im sorry if this is not the way to create a post but im new to github and im in a bit of hurry. Any help would be appreciated !
Thanks!

**Comment by miguelgrinberg** [comment, marked as answer]
You are using the polling transport, which is constantly sending requests to the server. If the WebSocket transport is available, the client will eventually upgrade to WebSocket, and there won't be any more requests like those you are showing.

**Reply by jcgomez11** [reply]
Thank you for the quick reply! I'll try specifying the transport when connecting the clients and see how it goes!
I'm running the server with eventlet like in the documentation ->  eventlet.wsgi.server(eventlet.listen(("0.0.0.0", 1234)), app, log = logger)
Currently the condition of the network is really bad, wifi connecting and disconnecting suddendly many times, and eventually the server thread is crashing, it stops accepting reconnections or any new connection. Any idea if the polling transport has any to do with the sever crashing? or is it the effect of many "ungraceful" disconnections going on?

**Reply by miguelgrinberg** [reply]
The polling transport should work. But if your wifi is unstable nothing is going to work well.
